### ISSUE TITLE
Don't break reports during 'project' report regeneration

### ISSUE THREAD

**Issue by rcampbel3** [exakat/exakat#129]
Hi,

First off - I love exakat! Thank you!

Here's an issue I've noticed: I have a very large and complex code repository that takes a very long time for me to run:

php ./exakat project -p $REPO

I'm running this on a 4CPU, 31GB VM, so... I've got some horsepower behind this, and it's still taking a long time...

The problem I'm facing is that while I'm running exakat on a project, it makes that report generate a 404 until the processing is complete.

Is there any way that new report processing could move from one atomic state to another so that we can continue to view the old report while the new report data is being generated?

**Comment by exakat** [comment]
Indeed, the 'project' command removes the report at once, while it could rely on the 'report' command to do this, when the report is almost ready to be published. That is a good idea.

As for keeping multiple reports, it is beyond exakat current set of command, as it would lead into managing report versions and storage. This is probably better as an external script, that steers exakat, and move the reports (with naming convention) into a good storage.

**Comment by rcampbel3** [comment]
I wasn't thinking about keeping multiple reports -- more like building a new report in a new directory, and only deleting/moving/changing symlink and cleaning up after the report finishes successfully. That would ensure that the reports on the web are always accessible, and should exakat project fail, at least the old data remains visible. You're also right -- I could script around this too, but I do think this would be desirable. Thanks
